g11n-ws/vip-manager-l10n-agent/src/main/java/com/vmware/l10agent/service/SingleComponentServiceImpl.java

```diff
@@ -213,52 +217,167 @@ private boolean synchkey2VipL10n(RecordModel record,  String key, String srcValu
         
         return true;
 	}
-	
-	
-	
-    /**
-     * sync the local source content to internal l10n or i18n
-     */
+
+
+
+	/**
+	 * sync the local source content to internal l10n or i18n
+	 */
 	@Override
 	public boolean synchComponentFile2Internal(RecordModel record) {
 
-      ComponentSourceModel model =  getSourceComponentFile(record);
-      if (model != null) {
-		  for (Entry<String, Object> entry : model.getMessages().entrySet()) {
-			  try {
-				  boolean result = synchkey2VipL10n(record, entry.getKey(), (String) entry.getValue(), null, null);
-				  if (!result) {
-					  result = synchkey2VipI18n(record, entry.getKey(), (String) entry.getValue(), null, null);
-					  if (!result) {
-						  try {
-							  if (record.getStatus() < 5) {
-								  record.setStatus(record.getStatus() + 1);
-								  TaskSysnQueues.SendComponentTasks.put(record);
-							  }
-						  } catch (InterruptedException e) {
-
-							  Thread.currentThread().interrupt();
-
-							  logger.info(e.getMessage(), e);
-						  }
-						  return result;
-					  }
-				  }
-			  } catch (UnsupportedEncodingException e) {
-
-				  logger.error(e.getMessage(), e);
-				  return false;
-			  }
-		  }
-	  }
-		return true;
+		ComponentSourceModel model =  getSourceComponentFile(record);
+		if (this.configs.isSyncBatchEnable()){
+			return syncBatchSource(record, model);
+		}else {
+		  return syncSingleSource(record, model);
+		}
+
 	}
-	
+
 	private String postData(String source, String urlStr)  throws VIPHttpException {
 		String response =  HTTPRequester.postData(source, urlStr, "application/json", "POST", null);
 		if(response == null || response.equalsIgnoreCase("")) {
 			throw new VIPHttpException("postFormData error.");
 		}
 		return response;
 	}
+
+	private boolean  syncSingleSource(RecordModel record, ComponentSourceModel model){
+		if (model != null) {
+			for (Entry<String, Object> entry : model.getMessages().entrySet()) {
+				try {
+					boolean result = syncKey2VipL10n(record, entry.getKey(), (String) entry.getValue());
+					if (!result) {
```

Comment:
if (!result) means when sync to l10n service fails, then sync to i18n service, why? Should it be sync to l10n success, then sync to i18n?

Reply:
It just send to i10n or 18n, not both